tests/test_pyrflx.py

```diff
@@ -586,16 +621,492 @@ def test_ethernet_parse_binary(self) -> None:
+            self.tlv_checksum.assign(test_bytes)
+
+    def test_generating_tlv_data(self) -> None:
+        expected = b"\x40\x04\x00\x00\x00\x00"
+
+        self.tlv.set("Tag", "Msg_Data")
+        self.tlv.set("Length", 4)
+        self.tlv.set("Value", b"\x00\x00\x00\x00")
+
+        self.assertTrue(self.tlv.valid_message)
+        self.assertEqual(self.tlv.value, expected)
+
+    def test_generating_tlv_data_zero(self) -> None:
+        """
+        expected = b"\x40\x00"
+        self.tlv_checksum.set("Tag", "Msg_Data")
+        self.tlv_checksum.set("Length", 0)
+
+        # Die Nachricht kann nicht valid werden, da Value nicht gesetzt werden
+        # kann, weil Length 0 ist
+        # self.tlv.set("Value", b"\x00\x00\x00\x00")
+
+        self.assertTrue(self.tlv_checksum.valid_message)
+        self.assertEqual(self.tlv_checksum.binary, expected)
+        """
+        raise NotImplementedError
+
+    def test_generating_tlv_error(self) -> None:
+
+        self.tlv_checksum.set("Tag", "Msg_Error")
+        self.assertTrue(self.tlv_checksum.valid_message)
+
+    def test_array_parse_form_bytes(self) -> None:
+
+        self.array_test_nested_msg.assign(b"\x02\x05\x06")
+        self.assertEqual(self.array_test_nested_msg.value, b"\x02\x05\x06")
+        self.array_test_typeval.assign(b"\x03\x05\x06\x07")
+        self.assertEqual(self.array_test_typeval.value, b"\x03\x05\x06\x07")
+
+    def test_array_nested_messages(self) -> None:
+
+        array_message_one = self.package_array_nested_msg["Foo"]
+        array_message_two = self.package_array_nested_msg["Foo"]
+
+        array_message_one.set("Byte", 5)
+        array_message_two.set("Byte", 6)
+
+        foos = [array_message_one, array_message_two]
+
+        self.array_test_nested_msg.set("Length", 2)
+        self.array_test_nested_msg.set("Bar", foos)
+
+        self.assertTrue(self.array_test_nested_msg.valid_message)
+        self.assertEqual(b"\x02\x05\x06", self.array_test_nested_msg.value)
+
+    def test_array_typevalues(self) -> None:
+
+        a = IntegerValue(model.ModularInteger("Array_Type.Byte_One", Number(256)))
+        b = IntegerValue(model.ModularInteger("Array_Type.Byte_Two", Number(256)))
+        c = IntegerValue(model.ModularInteger("Array_Type.Byte_Three", Number(256)))
+        a.assign(5)
+        b.assign(6)
+        c.assign(7)
+
+        byte_array = [a, b, c]
+
+        self.array_test_typeval.set("Length", 3)
+        self.array_test_typeval.set("Byte", byte_array)
+
+        self.assertTrue(self.array_test_typeval.valid_message)
+        self.assertEqual(self.array_test_typeval.value, b"\x03\x05\x06\x07")
+
+    def test_arrayvalue(self) -> None:
+        # pylint: disable=protected-access
+
+        type_array = ArrayValue(
+            Array("Test.Array", model.ModularInteger("Test.Mod_Int", Number(256)))
+        )
+        type_array._is_message_array = False
+        type_array._element_type = model.ModularInteger("Test.Mod_Int", Number(256))
```

Comment:
Why do you need to set these protected properties here? Shouldn't that do the constructor of ArrayValue?

Reply:
You are right, it does that now after the changes of 192 were merged and i refactored the constructor to utilize them. I removed the affected lines.